src/ripple/app/paths/impl/XRPEndpointStep.cpp

```diff
@@ -273,7 +283,20 @@ make_XRPEndpointStep (
     StrandContext const& ctx,
     AccountID const& acc)
 {
-    auto r = std::make_unique<XRPEndpointStep> (acc, ctx.isLast, ctx.j);
+    // For historical reasons, offer crossing is allowed to dig further
+    // into the XRP reserve than an ordinary payment.  (I believe it's
+    // because the trustline was created after the XRP was removed.)  Pass
+    // a flag that allows us to simulate the historical behavior.
+    //
+    // Note that reduced reserve only happens if the trustline does not
+    // currently exist.
+    bool reduceReserve = ctx.offerCrossing && ctx.isFirst;
+    if (reduceReserve)
+        reduceReserve = !ctx.view.read (keylet::line (acc, ctx.strandDeliver));
```

Comment:
I'd like to get rid of this rule on the grounds it complicates the code for not enough benefit. If we remove this, we no longer have to keep `strandDeliver` in the ctx and we simplify XRPEndpointStep. Of course, a rule change like this should be discussed with up the chain of command.